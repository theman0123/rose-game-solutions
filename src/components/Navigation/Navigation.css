.Link {
  color: black;
  font-size: 8vmin;
  margin: 0.5em 0.5em 0 0.5em;
  text-decoration: none;
}

.Container-horizontal {
  display: flex;
  flex-direction: row;
}

.Nav-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
