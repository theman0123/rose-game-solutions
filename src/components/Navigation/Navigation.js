import React from "react";
import { Link } from "react-router-dom";
import "./Navigation.css";
import { routes } from "../../routes/routes.config";

// don't show current route in navigation
const filteredRoutes = (iterate, currentLocation) =>
  iterate.filter(route => route.to !== currentLocation);

export default props => {
  return (
    <div className='Nav-container'>
      <div className='Container-horizontal'>
        {filteredRoutes(routes, props.history.location.pathname).map(route => (
          <Link className='Link' to={route.to} key={route.name}>
            {route.name}
          </Link>
        ))}
      </div>
      <div className='Current-location'>
        {
          routes.filter(
            route => route.to === props.history.location.pathname,
          )[0].name
        }
      </div>
    </div>
  );
};
