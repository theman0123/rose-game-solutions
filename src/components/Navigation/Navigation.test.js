import React from "react";
import renderer from "react-test-renderer";
import { Route, MemoryRouter } from "react-router-dom";
import Navigation from "./Navigation";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Route to='/' component={Navigation} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("should not render link of current route", () => {
  const tree = renderer
    .create(
      <MemoryRouter
        initialEntries={["/games", "/about", "/", "/services"]}
        initialIndex={0}
      >
        <Route to='/' component={Navigation} />
      </MemoryRouter>,
    )
    .toJSON();
  // route total is 4, should be 3 after filtering for current location
  // help: tree === first <div>
  expect(tree.children[0].children.length).toEqual(3);
});
