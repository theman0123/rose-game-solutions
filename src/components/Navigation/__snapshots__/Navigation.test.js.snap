// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Nav-container"
>
  <div
    className="Container-horizontal"
  >
    <a
      className="Link"
      href="/games"
      onClick={[Function]}
    >
      Games
    </a>
    <a
      className="Link"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="Link"
      href="/services"
      onClick={[Function]}
    >
      Services
    </a>
  </div>
  <div
    className="Current-location"
  >
    Home
  </div>
</div>
`;
